src/main/java/com/aws/greengrass/shadowmanager/ipc/InboundRateLimiter.java

```diff
@@ -71,22 +103,18 @@ public void clear() {
     }
 
     /**
-     * Sets the overall inbound rate limiter rate.
-     *
-     * @param rate Max inbound requests per second for all things
+     * get total inbound rate limiter.
+     * @return totalInboundRateLimiter
      */
-    public void setTotalRate(int rate) {
-        totalRate.set(rate);
-        totalInboundRateLimiter.setRate(totalRate.get());
+    public IpcRateLimiter getTotalInboundRateLimiter() {
+        synchronized (totalInboundRateLimiter) {
+            totalInboundRateLimiter.setRate(getTotalRate());
+        }
+        return totalInboundRateLimiter;
     }
 
-    /**
-     * Sets each inbound rate limiter per thing to specified rate.
-     *
-     * @param rate Max inbound requests per second per thing
-     */
-    public void setRate(int rate) {
-        ratePerThing.set(rate);
-        rateLimitersPerThing.forEach((k, v) -> v.setRate(ratePerThing.get()));
+    public Map<String, IpcRateLimiter> getRateLimitersPerThing() {
+        rateLimitersPerThing.forEach((k, v) -> v.setRate(getRatePerThing()));
```

Comment:
does `setRate` need to be called on config changes? If so, how are we doing that?